packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -144,16 +154,15 @@ export async function generateBlogPosts(
 
       // Use file create time for blog.
       date = date || (await fs.stat(source)).birthtime;
+
+      const slug =
+        frontMatter.slug || (match ? toUrl({date, link: linkName}) : linkName);
       frontMatter.title = frontMatter.title || linkName;
 
       blogPosts.push({
-        id: frontMatter.id || frontMatter.title,
+        id: frontMatter.slug || frontMatter.title,
         metadata: {
-          permalink: normalizeUrl([
-            baseUrl,
-            routeBasePath,
-            frontMatter.id || toUrl({date, link: linkName}),
-          ]),
+          permalink: normalizeUrl([baseUrl, routeBasePath, formatSlug(slug)]),
```

Comment:
I think slug should be optional. The title can be used as the path in that case.